BUILD.md

```diff
@@ -0,0 +1,118 @@
+# cuML Build From Source Guide
+
+## Setting Up Your Build Environment
+
+To install cuML from source, ensure the dependencies are met:
+
+1. [cuDF](https://github.com/rapidsai/cudf) (>=0.7)
+2. zlib
+3. cmake (>= 3.12.4)
+4. CUDA (>= 9.2)
+5. Cython (>= 0.29)
+6. gcc (>=5.4.0)
+7. BLAS - Any BLAS compatible with cmake's [FindBLAS](https://cmake.org/cmake/help/v3.12/module/FindBLAS.html). Note that the blas has to be installed to the same folder system as cmake, for example if using conda installed cmake, the blas implementation should also be installed in the conda environment.
+
+### Installing from Source:
+
+Once dependencies are present, follow the steps below:
+
+1. Clone the repository.
+```bash
+$ git clone --recurse-submodules https://github.com/rapidsai/cuml.git
+```
+
+2. Build and install `libcuml` (the C++/CUDA library containing the cuML algorithms), starting from the repository root folder:
+```bash
+$ cd cuML
+$ mkdir build
+$ cd build
+$ export CUDA_BIN_PATH=$CUDA_HOME # (optional env variable if cuda binary is not in the PATH. Default CUDA_HOME=/path/to/cuda/)
+$ cmake ..
+```
+
+If using a conda environment (recommended), then cmake can be configured appropriately via:
+
+```bash
+$ cmake .. -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX
+```
+
+Additionally while using conda and a conda installed cmake, the `openblas` conda package can be explicitly specified for `blas` and `lapack`:
+
+```bash
+cmake .. -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX -DBLAS_LIBRARIES=$CONDA_PREFIX/lib/libopenblas.a -DLAPACK_LIBRARIES=$CONDA_PREFIX/lib/libopenblas.a
```

Comment:
We discussed this, but just capturing for a review. I would use shared libs here to be safe.